tests/lib/rules/forbid-component-props.js

```diff
@@ -143,11 +196,181 @@ ruleTester.run('forbid-component-props', rule, {
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', 'style'], defaultAllowOnDOM: false}],
+    errors: [
+      {
+        message: DOM_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', allow: ['div']}], defaultAllowOnDOM: false}],
+    errors: [
+      {
+        message: DOM_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', forbid: ['i']}]}],
+    errors: [
+      {
+        message: TAG_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 20,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo className="foo"><i style={{color: "red"}} /><div style={{color: "red"}} /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: ['className', {property: 'style', forbid: ['i']}]}],
+    errors: [
+      {
+        message: CLASSNAME_ERROR_MESSAGE,
+        line: 4,
+        column: 17,
+        type: 'JSXAttribute'
+      }, {
+        message: TAG_STYLE_ERROR_MESSAGE,
+        line: 4,
+        column: 36,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
+      '  propTypes: externalPropTypes,',
+      '  render: function() {',
+      '    return <Foo className="foo"><i className="foo" /></Foo>;',
+      '  }',
+      '});'
+    ].join('\n'),
+    parserOptions: parserOptions,
+    options: [{forbid: [{property: 'className', forbid: ['i']}], defaultAllowOnComponents: true}],
+    errors: [
+      {
+        message: TAG_CLASSNAME_ERROR_MESSAGE,
+        line: 4,
+        column: 36,
+        type: 'JSXAttribute'
+      }
+    ]
+  }, {
+    code: [
+      'var First = React.createClass({',
```

Comment:
I only added one total test case for each. I figure if one test case works for all three, all test cases should, because we're only getting at the JSX tags inside the (render) function body, right?  I can add more if you think they're necessary.